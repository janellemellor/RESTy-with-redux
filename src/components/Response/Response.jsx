import React from 'react';

const Response = () => {
  
    return (
      <pre>
        {JSON.stringify(response, null, 2)}
      </pre>
    );
};

export default Response;
