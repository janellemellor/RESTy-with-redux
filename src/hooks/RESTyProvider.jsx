import React, { createContext, useState, useContext } from 'react';

const RESTyContext = createContext();

export const RESTyProvider = ({ children }) => {
  const [url, setUrl] = useState('');
  const [method, setMethod] = useState('GET');
  const [jsonBody, setJsonBody] = useState('');
  
  const onChange = ({ target }) => {
    const setInputsFactory = {
      url: setUrl, 
      method: setMethod, 
      jsonBody: setJsonBody
    };
    setInputsFactory[target.name](target.value);
  };

  const context = {
    url,
    method, 
    jsonBody, 
    onChange
  };

  return (
    <RESTyContext.Provider value={context} >
      {children}
    </RESTyContext.Provider>
  );
};

export const useRESTy = () => {
  const context = useContext(RESTyContext);
  return context;
};
