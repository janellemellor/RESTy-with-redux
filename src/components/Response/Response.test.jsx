import React from 'react';
import { shallow } from 'enzyme';
import Response from './Response';
import { RESTyProvider } from '../../hooks/RESTyProvider';

describe('Response Component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <RESTyProvider>
        <Response response='{}' />
      </RESTyProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });  
});
