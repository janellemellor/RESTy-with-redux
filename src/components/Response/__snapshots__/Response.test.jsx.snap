// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response Component matches a snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "jsonBody": "",
      "method": "GET",
      "onChange": [Function],
      "onSubmit": [Function],
      "url": "",
    }
  }
>
  <Response
    response="{}"
  />
</ContextProvider>
`;
